### PR TITLE
Build rsm production

### PR DESCRIPTION
**Pull request by neillturner**
### Context

Create production environment as part of rsm migration 

### Changes proposed in this pull request

Create production sizing for database and application.
No logging or monitoring initially so no conflict with existing non-aks application

### Guidance to review

check code for anything that needs changing for production that I have incorrect. 
you can run a plan 
make aks-production aks-terraform-plan

when happy i will apply 

### Link to Trello card

[RSM: build production environment](https://trello.com/c/vLHbTudZ/2103-rsm-build-production-environment)

### Checklist

- [ ] Attach to Trello card
- [ ] Rebased main
- [ ] Cleaned commit history
- [ ] Tested by running locally

### CODE REVIEW

**Comment by neillturner** [comment]
[RSM: build production environment](https://trello.com/c/vLHbTudZ/2103-rsm-build-production-environment)

File: terraform/application/config/production.tfvars.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "app_replicas": 2,
+  "worker_replicas": 1,
```
Comment (RMcVelia) [inline terraform/application/config/production.tfvars.json:3]: have you confirmed with the app team it is safe to run a worker on a copy of prod?

File: terraform/application/config/production.tfvars.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+  "app_replicas": 2,
```
Comment (RMcVelia) [inline terraform/application/config/production.tfvars.json:2]: Do we need basic-auth initially to stop anyone using the prod copy?

File: terraform/application/config/production.tfvars.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  "worker_replicas": 1,
+  "postgres_flexible_server_sku": "GP_Standard_D2ds_v4",
+  "enable_postgres_backup_storage": true,
+  "postgres_enable_high_availability": true,
+  "send_traffic_to_maintenance_page": false,
+  "account_replication_type": "GRS",
+  "cluster": "production",
+  "enable_logit": false,
```
Comment (RMcVelia) [inline terraform/application/config/production.tfvars.json:10]: I would think it's ok to enable logit so any issues can be investigated.

File: terraform/application/database.tf [hunk 4]
```diff
@@ -1,18 +1,20 @@
...
+  config_short                   = var.config_short
+  cluster_configuration_map      = module.cluster_data.configuration_map
+  azure_extensions               = ["PGCRYPTO"]
+  use_azure                      = var.deploy_azure_backing_services
+  azure_enable_monitoring        = var.enable_monitoring
+  azure_enable_backup_storage    = var.enable_postgres_backup_storage
+  azure_sku_name                 = var.postgres_flexible_server_sku
+  azure_enable_high_availability = var.postgres_enable_high_availability
+  server_version                 = "16"
 }
```
Comment (RMcVelia) [inline terraform/application/database.tf:18]: maintenance window for prod as per https://github.com/DFE-Digital/teacher-services-cloud/blob/main/documentation/production-checklist.md#maintenance-window

File: terraform/application/config/production.tfvars.json (hunk 1 above)
Comment (neillturner) [reply terraform/application/config/production.tfvars.json:3]: don't know

File: terraform/application/config/production.tfvars.json (hunk 2 above)
Comment (neillturner) [reply terraform/application/config/production.tfvars.json:2]: don't know

File: terraform/application/config/production.tfvars.json (hunk 3 above)
Comment (neillturner) [reply terraform/application/config/production.tfvars.json:10]: ok

File: terraform/application/database.tf (hunk 4 above)
Comment (neillturner) [reply terraform/application/database.tf:18]: ok

File: terraform/application/config/production.tfvars.json (hunk 2 above)
Comment (neillturner) [reply terraform/application/config/production.tfvars.json:2]: we do have SUPPORT-USER and SUPPORT-PASSWORD set in thew keyvault so in prod you will get s login popup just like in preprod https://[preprod.refer-serious-misconduct.education.gov.uk](https://preprod.refer-serious-misconduct.education.gov.uk/)/

File: terraform/application/config/production.tfvars.json (hunk 1 above)
Comment (neillturner) [reply terraform/application/config/production.tfvars.json]: set worker replicas to 0 so should not be any workers initially

File: terraform/application/database.tf (hunk 4 above)
Comment (RMcVelia) [reply terraform/application/database.tf:18]: I don't see a maintenance window for prod postgres?

Comment (neillturner) [reply terraform/application/database.tf:19]: added azure maintenance window for postgres to 2am

**Review by RMcVelia — APPROVED** [review]

**Pull request merged**